backend/src/models/reports/Tab_49_rpt_PF_NetRecoveries.js

```diff
@@ -2,13 +2,202 @@ const dbConnection = require("@database/databaseConnection");
+      current_year_recoveries_stob_no_param.project_number,
+      current_year_recoveries_stob_no_param.project_name,
+      current_year_recoveries_stob_no_param.portfolio_id,
+      current_year_recoveries_stob_no_param.portfolio_name,
+      current_year_recoveries_stob_no_param.total_project_budget,
+      current_year_recoveries_stob_no_param.fiscal_year AS fy,
+      current_year_recoveries_stob_no_param.fiscal,
+      financial_recovery_by_portfolio_and_stob_recovered_no_param.contract_costs AS contract_recovered,
+      financial_recovery_by_portfolio_and_stob_recovered_no_param.travel_costs AS travel_recovered,
+      financial_recovery_by_portfolio_and_stob_recovered_no_param.business_expenses AS business_expense_recovered,
+      financial_recovery_by_portfolio_and_stob_recovered_no_param.staff_recoveries AS staff_recoveries_recovered,
+      financial_recovery_by_portfolio_and_stob_recovered_no_param.other AS other_recovered,
+      SUM(
+        current_year_recoveries_stob_no_param.current_fy_recovered_to_date
+      ) AS cy_recovered_to_date,
+      SUM(
+        current_year_recoveries_stob_no_param.current_fy_total_recoverable
+      ) AS cy_total_recoverable
+    FROM financial_recovery_by_portfolio_and_stob_recovered_no_param
+      INNER JOIN current_year_recoveries_stob_no_param ON (
+        financial_recovery_by_portfolio_and_stob_recovered_no_param.fiscal_year = current_year_recoveries_stob_no_param.fiscal_year
+      )
+      AND (
+        financial_recovery_by_portfolio_and_stob_recovered_no_param.portfolio_id = current_year_recoveries_stob_no_param.portfolio_id
+      )
+      AND (
+        financial_recovery_by_portfolio_and_stob_recovered_no_param.project_id = current_year_recoveries_stob_no_param.project_id
+      ) 
+      GROUP BY
+      current_year_recoveries_stob_no_param.project_id,
+      current_year_recoveries_stob_no_param.project_number,
+      current_year_recoveries_stob_no_param.project_name,
+      current_year_recoveries_stob_no_param.portfolio_id,
+      current_year_recoveries_stob_no_param.portfolio_name,
+      current_year_recoveries_stob_no_param.portfolio_abbrev,
+      current_year_recoveries_stob_no_param.total_project_budget,
+      total_project_budget - recoverable_amount,
+      current_year_recoveries_stob_no_param.fiscal_year,
+      current_year_recoveries_stob_no_param.fiscal,
+      financial_recovery_by_portfolio_and_stob_recovered_no_param.contract_costs,
+      financial_recovery_by_portfolio_and_stob_recovered_no_param.travel_costs,
+      financial_recovery_by_portfolio_and_stob_recovered_no_param.business_expenses,
+      financial_recovery_by_portfolio_and_stob_recovered_no_param.staff_recoveries,
+      financial_recovery_by_portfolio_and_stob_recovered_no_param.other
+    ORDER BY current_year_recoveries_stob_no_param.portfolio_name,
+      current_year_recoveries_stob_no_param.project_number
+    ) as q`
+    );
+
+  handleParams(query, requestParams);
```

Comment:
the query works

Reply:
I might try to clean things up after the template is done, if I have the chance. Thanks!